frontend/src/pages/TaskOverviewPage.tsx

```diff
@@ -164,12 +164,18 @@ export const getTaskOverviewData = (
   return { metrics, data };
 };
 
+interface LocationState {
+  from: string | undefined;
+}
+
 const TaskOverview: FC<{
   tasks: Task[];
   task: Task;
   taskIdx: number;
 }> = ({ tasks, task, taskIdx }) => {
   const history = useHistory();
+  const location = useLocation<LocationState>();
+  const fromUrl = (location.state as LocationState | undefined)?.from;
```

Comment:
```suggestion
  const { state } = useLocation<LocationState | undefined>();
```

This could be simplified a bit.

And `undefined` can be added to the type to avoid the typecast later.